### DISCUSSION TITLE
SAFE procedure + heterogenous terminology + 100K records + (<0.04%) anticipated relevant records

### DISCUSSION THREAD

**Discussion by Emanuel-1986** [asreview/asreview#1780] [Q&A]
Hi,

I am currently designing the protocol for a systematic review of prognostic model studies. Unfortunately, the terminology in this field of medicine is very heterogeneous. Despite using validated search filters, my preliminary PubMed search retrieved 50K, Embase retrieved 57K, and CINAHL retrieved 23K records. I still have to execute the searches, but upon merging and deduplicating the datasets, I suspect I will end up with a dataset of ~100K records. Therefore, I will use ASReview.

Despite conducting a preliminary search in PubMed, I am not aware of any prognostic model specifically developed for the condition that I am investigating. Therefore, apart from searching for prognostic models evaluating the specific primary condition, I plan to search for prognostic models developed for conditions that are highly associated with the primary condition that I am studying (I call these secondary conditions). Finding models for both the primary and secondary conditions will give me an idea about the R2 values for sample size calculation when designing a prospective cohort study and give an idea of the included predictors within these models. I anticipate the number of relevant records in my search to be <0.04%, mainly due to finding models of the secondary conditions. 

Below is my plan for pragmatically adopting the SAFE procedure:
1.	SAFE phase 1 - Generate 450 random records. 2 independent reviewers will screen the same 450 records. If the inter-rater agreement is less than 80%, the sources of discrepancies will be analysed and discussed, and the calibration exercise will be repeated until an acceptable level of inter-rater agreement is achieved. Once we reach the 80% agreement, any identified relevant records will be discussed and we will mutually decide if they are relevant, hence having a final parsimonious dataset and then using this dataset for phase 2. If we reach the 80% agreement but fail to identify a single relevant record, we will mark all the articles I previously earmarked as prior knowledge in the large dataset (100K). This can include my personal bias, but otherwise, I would spend too much time screening a random subset of the data, hoping to find a relevant one.  I will use the code below to generate 450 random records (as stated in: https://github.com/asreview/asreview-datatools?tab=readme-ov-file#sample)

asreview data sample input_dataset.xlsx output_dataset.xslx 150

2.	SAFE phase 2 - I plan to use LR + TF-IDF since even though this classifier and feature extractor came second in Teijema et al. (2023) (Table 1), the feature extractor time (23sec) was much, much lower compared to the LR+ SBERT (6hrs) for a dataset that is nearly half (46K) the one that I will screen (~100K)! In this phase, initially, I plan to screen a further 7,500 titles. However, phase 2 of the SAFE procedure provides 4 bullets illustrating the stopping heuristic; beneath them, within the text, it is mentioned that by inspecting the recall plot and noticing a plateau, the remaining stopping rules can be checked. Therefore, I was thinking of stopping this phase if no relevant articles will be identified after screening 1,000 consecutive records and hence having a plateau on the recall plot.

3.	SAFE phase 3 - model switching. Jelle kindly suggested using doc2vec, sBERT, or ScholarBERT. However, considering the potentially large size of the dataset and background information, I would appreciate your advice on which model to use in this phase. Regarding the stopping heuristic, I was going to abide by the SAFE suggestion of stopping after screening 50 consecutive irrelevant titles.

4.	Then, for phase 4, which suggests using a simple model, should I use again the LR + TF-IDF or switch to the Naïve Bayes + TF-IDF?

In GitHub, it was mentioned that you were working on multilingual models for ASReview. Is the paper and model out?

I look forward to your response. 

Thank you
Emanuel

**Comment by Rensvandeschoot** [comment]
1) It sounds like a good plan!

2) Stopping after 1.000 consecutive irrelevant records looks pretty conservative. I would suggest a lower number for this phase combined with a slightly higher number after the complex model (phase 3) so that you screen 1.000 irrelevant records in a row, for example, 500 and 500. But I have seen many papers with even smaller numbers. It is up to you how conservative you want to be.

3) So far, for the models available in ASReview, we have seen a good performance of SBERT, and you can combine it with the 2NN classifier. If you are able to add new models via the [template](https://github.com/asreview/template-extension-new-model), you even have more options to use any model you would like. @jteijema has prepared a collection of such models in https://github.com/jteijema/ASReview-experimental-models, including multi-language ones. 

4) yes, indeed, a simple model will do

The paper, comparing different multilanguage models, is close to being accepted. The pre-print is available [here](https://osf.io/preprints/psyarxiv/zht9d).

**Comment by Emanuel-1986** [comment]
Dear Prof van de Schoot,

I read with great interest the pre-print entitled “Advancing multilingual abstract classification: a comparative analysis of feature extraction models in systematic reviews.” It mentioned that the MUSE, LaBSE, and mlongT5 models were the best of those evaluated. However, MUSE and LaBSE were less computationally demanding. 

1.	Which one would you choose?
2.	Approximately how long were the feature extractor time and the median iteration time for each of these models when using the logistic regression as a classifier? Is it in the range of 6 hours or 30 minutes? 
3.	Do you have any suggestions on integrating multi-language models into the SAFE procedure? 

Within my PubMed search, out of 50K records, 850 records (1.7%) did not match the English language filter in PubMed. 

4.	I considered including a multilingual screen between phases 3 and 4 of the SAFE procedure. Hence the records labelled from the previous three phases will serve as prior knowledge. What do you think? 
5.	However, since such prior knowledge is all in the English language, could that bias the multilanguage model?

On the assessment of inter-rater reliability when using ASReview:

6.	Do you have any suggestions? 
7.	After finishing each phase of the SAFE procedure, should the reviewers check for agreement on the records labelled as relevant?
8.	Does it make sense that at the end of each phase of the SAFE procedure, i.e., after the stopping rule elapses, the two raters discuss and reach consensus on which records are labelled as relevant and then re-label the mismatched records so that at the beginning of phases 2, 3, and 4, the records labelled as relevant are the same for both reviewers? 

Thank you for your time and guidance. 

Kind regards,
Emanuel

**Reply by jteijema** [reply]
Great questions! We ran some simulations on our new SYNERGY dataset. I think I can answer 1 2 and 3:

1. LaBSE! To us, the performance seemed best when evaluated taking into account the computational time. Check out it's interplay with classifiers at [our results website](https://jteijema.github.io/synergy-simulations-website/models.html#labse).

2. The LaBSE embedding computation speed depends a lot on the size of dataset. But expect a few hours for a large dataset on a laptop. Iteration time depends on which classifier you pick, ranging from 0.02 seconds to around 7 seconds. 

3. Usually switching to a more complex FE is a good choice, but for multi lingual datasets it might become difficult if the starting FE is not capable of different languages. I'd recommend sticking to LaBSE in this case!

Jelle

**Reply by bethgr** [reply]
Regarding IRR and questions 6, 7, and 8:

6. There is a helpful ASReview report tool created by Sam Langton. You can view an example of the report here: https://rpubs.com/langton_/asreview_report_example. Links to the repository and instructions can also be found on Sam's page.

7. and 8. Yes, I recommend conducting these checks and recalibrations between raters along the way, as you suggested.

Beth

**Reply by jteijema** [reply]
Actually I have an opinion on 4 and 5. If the multilingual model works as advertised, then language should not be encoded as a detectable feature in the embedding. That's a big _if_, but the multilingual model we're planning on implementing is one of the highest quality models around. I'd personally suggest using the model in the manner you describe.
